packages/e2e-test-utils/src/templates.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { rest, batch } from './rest-api';
+
+const PATH_MAPPING = {
+	wp_template: '/wp/v2/templates',
+	wp_template_part: '/wp/v2/template-parts',
+};
+
+/**
+ * Delete all the templates of given type.
+ *
+ * @param {('wp_template'|'wp_template_part')} type - Template type to delete.
+ */
+export async function deleteAllTemplates( type ) {
+	const path = PATH_MAPPING[ type ];
+
+	if ( ! path ) {
+		return;
+	}
+
+	const [ templates ] = await rest( { path } );
```

Comment:
Something to look out for here (that also caught me out recently) is that you might need `per_page=-1` to ensure all templates are definitely removed.

Reply:
I thought that was only true when working with the entities and not direct API calls. Templates API isn't paginated.

Reply:
Yep, you're right, it doesn't seem to be paginated. Looks like the custom REST controller that templates use doesn't implement it.